stage1/init/init.go

```diff
@@ -588,12 +576,39 @@ func stage1(rp *stage1commontypes.RuntimePod) int {
 		diag.Printf,
 	)
 
+	// set hostname inside pod
+	// According to systemd manual (https://www.freedesktop.org/software/systemd/man/hostname.html) :
+	// "The /etc/hostname file configures the name of the local system that is set
+	// during boot using the sethostname system call"
+	if p.Hostname == "" {
+		p.Hostname = stage1initcommon.GetMachineID(p)
+	}
+	hostnamePath := filepath.Join(common.Stage1RootfsPath(p.Root), "etc/hostname")
+	if err := ioutil.WriteFile(hostnamePath, []byte(p.Hostname), 0644); err != nil {
+		log.FatalE("error writing "+hostnamePath, err)
```

Comment:
Good catch, though I was just matching style. This whole file needs to be fixed; I can take care of that in another PR.